services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy

```diff
@@ -212,6 +215,68 @@ class SwitchSyncSpec extends BaseSpecification {
+        def flow = flowHelper.randomFlow(switchPair)
+        flowHelper.addFlow(flow)
+
+        and: "Reproduce situation when switches have missing rules and meters"
+        def involvedSwitches = pathHelper.getInvolvedSwitches(flow.id)
+        def cookiesMap = involvedSwitches.collectEntries { sw ->
+            [sw.dpId, northbound.getSwitchRules(sw.dpId).flowEntries.findAll {
+                !(it.cookie in sw.defaultCookies)
+            }*.cookie]
+        }
+        def metersMap = involvedSwitches.collectEntries { sw ->
+            [sw.dpId, northbound.getAllMeters(sw.dpId).meterEntries.findAll {
+                it.meterId > MAX_SYSTEM_RULE_METER_ID
+            }*.meterId]
+        }
+
+        involvedSwitches.each { northbound.deleteSwitchRules(it.dpId, DeleteRulesAction.IGNORE_DEFAULTS) }
+        [switchPair.src, switchPair.dst].each { northbound.deleteMeter(it.dpId, metersMap[it.dpId][0]) }
+        Wrappers.wait(RULES_DELETION_TIME) {
+            def validationResultsMap = involvedSwitches.collectEntries { [it.dpId, northbound.validateSwitch(it.dpId)] }
+            involvedSwitches.each { assert validationResultsMap[it.dpId].rules.missing.size() == 2 }
+            [switchPair.src, switchPair.dst].each { assert validationResultsMap[it.dpId].meters.missing.size() == 1 }
+        }
+
+        when: "Try to synchronize all switches"
+        def syncResultsMap = involvedSwitches.collectEntries { [it.dpId, northbound.synchronizeSwitch(it.dpId, false)] }
+
+        then: "System detects missing rules and meters, then installs them"
+        involvedSwitches.each {
+            assert syncResultsMap[it.dpId].rules.proper.size() == 0
+            assert syncResultsMap[it.dpId].rules.excess.size() == 0
+            assert syncResultsMap[it.dpId].rules.missing.containsAll(cookiesMap[it.dpId])
+            assert syncResultsMap[it.dpId].rules.removed.size() == 0
+            assert syncResultsMap[it.dpId].rules.installed.containsAll(cookiesMap[it.dpId])
+        }
+        [switchPair.src, switchPair.dst].each {
+            assert syncResultsMap[it.dpId].meters.proper.size() == 0
+            assert syncResultsMap[it.dpId].meters.excess.size() == 0
+            assert syncResultsMap[it.dpId].meters.missing*.meterId == metersMap[it.dpId]
+            assert syncResultsMap[it.dpId].meters.removed.size() == 0
+            assert syncResultsMap[it.dpId].meters.installed*.meterId == metersMap[it.dpId]
+        }
+
+        and: "Switch validation doesn't complain about missing rules and meters"
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            involvedSwitches.each {
+                def validationResult = northbound.validateSwitch(it.dpId)
+                assert validationResult.rules.missing.size() == 0
+                assert validationResult.meters.missing.size() == 0
+            }
+        }
+
+        and: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+    }
```

Comment:
You are not verifying that the 'synced' rules are indeed vxlan rules. This is fine, but in this case I want to be sure that our 'switch validation' is capable of detecting wrong vxlan rules (e.g. required rule is vxlan, but in fact installed is vlan, validation should complain)

Reply:
will do it when pr #2503 is merged